python-sdk/nuimages/nuimages.py

```diff
@@ -238,108 +298,521 @@ def list_categories(self, image_tokens: List[str] = None) -> None:
+
+                draw.bitmap((0, 0), Image.fromarray(mask * 128), fill=tuple(color + (128,)))
+
+            # Load object instances.
+            object_anns = [o for o in self.object_ann if o['sample_data_token'] == sd_token_camera]
+            if object_tokens is not None:
+                object_anns = [o for o in object_anns if o['token'] in object_tokens]
+
+            # Draw object instances.
+            for ann in object_anns:
+                # Get color, box, mask and name.
+                category_token = ann['category_token']
+                category_name = self.get('category', category_token)['name']
+                color = self.color_map[category_name]
+                bbox = ann['bbox']
+                attr_tokens = ann['attribute_tokens']
+                attributes = [self.get('attribute', at) for at in attr_tokens]
+                name = annotation_name(attributes, category_name, with_attributes=with_attributes)
+                if ann['mask'] is None:
+                    continue
+                mask = mask_decode(ann['mask'])
+
+                # Draw rectangle, text and mask.
+                draw.rectangle(bbox, outline=color)
+                if with_category:
+                    draw.text((bbox[0], bbox[1]), name, font=font)
+                draw.bitmap((0, 0), Image.fromarray(mask * 128), fill=tuple(color + (128,)))
 
         # Plot the image.
-        if ax is None:
-            _, ax = plt.subplots(1, 1, figsize=(9 * render_scale, 16 * render_scale))
-        ax.imshow(im)
         (width, height) = im.size
-        ax.set_xlim(0, width)
-        ax.set_ylim(height, 0)
-        ax.set_title(image_token)
-        ax.axis('off')
+        pix_to_inch = 100
+        figsize = (height / pix_to_inch, width / pix_to_inch)
+        plt.figure(figsize=figsize)
+        plt.axis('off')
+        plt.imshow(im)
+
+        # Save to disk.
+        if out_path is not None:
+            plt.savefig(out_path, bbox_inches='tight', dpi=2.295 * pix_to_inch * render_scale, pad_inches=0)
+            plt.close()
+
+    def render_depth(self,
+                     sd_token_camera: str,
+                     mode: str = 'sparse',
+                     max_depth: float = None,
+                     cmap: str = 'viridis',
+                     render_scale: float = 1.0,
+                     out_path: str = None) -> None:
+        """
+        This function plots an image and its depth map, either as a set of sparse points, or with depth completion.
+        Default depth colors range from yellow (close) to blue (far). Missing values are blue.
+        Suitable colormaps for depth maps are viridis and magma.
+        :param sd_token_camera: The sample_data token of the camera image.
+        :param mode: How to render the depth, either sparse or dense.
+        :param max_depth: The maximum depth used for scaling the color values. If None, the actual maximum is used.
+        :param cmap: The matplotlib color map name.
```

Comment:
Do we want to restrict users to just viridis and magma for the cmap? If yes, then perhaps an assertion statement somewhere?

Reply:
I've added a note, but will leave it to the user to decide what matlab colormap is best.